xrpl/models/transactions/pseudo_transactions/set_fee.py

```diff
@@ -19,42 +20,138 @@ class SetFee(PseudoTransaction):
     <https://xrpl.org/transaction-cost.html>`_ or `reserve requirements
     <https://xrpl.org/reserves.html>`_ as a result of `Fee Voting
     <https://xrpl.org/fee-voting.html>`_.
+
+    The parameters are different depending on if this is before or after the
+    `XRPFees Amendment<https://xrpl.org/known-amendments.html#xrpfees>`_
+
+    Before the XRPFees Amendment which was proposed in rippled 1.10.0
+    base_fee, reference_fee_units, reserve_base, and reserve_increment
+    were required fields.
+
+    After the XRPFees Amendment, base_fee_drops, reserve_base_drops,
+    and reserve_increment_drops are required fields.
+
+    No SetFee Psuedo Transaction should contain fields from BOTH before
+    and after the XRPFees amendment.
     """
 
-    base_fee: str = REQUIRED  # type: ignore
+    def __post_init__(self: SetFee) -> None:
```

Comment:
I don't think these checks are needed.  A consumer can never create the `SetFee` transaction or change the Fees ledger object.

Reply:
Tried doing that just now, that causes problems when trying to use `from_xrpl` / `from_dict` because it tries to look up the `SetFee` transaction, but just finds the type alias and can't find the function definition. I think the optional parameters are good enough here since I think `from_dict` might be involved in parsing even if we're just reading data from the ledger.